CONTRIBUTING.md

```diff
@@ -3,62 +3,75 @@
+Besides language runtimes, package managers, and command line programs, tooling should be written to assume that libraries are installed within the package, not globally available on the system. I.e, do not asusme the `eslint` or `prettier` are installed globally.
 
-### AWS
+## Getting Started
 
-There are a number of shell scripts in this repo. They assume you have a profile in `~/.aws/config` called `prx-legacy`, which is configured to access the `prx-legacy` AWS account.
+After cloning the Git repository for Porter, run `make bootstrap` to install all necessary development dependencies. This requires already having Node.js, Ruby, and Python 3 installed on your system.
 
 ## Deploying
 
-The template is intended to deployed using the [SAM CLI](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-sam-cli-command-reference.html). You can run `sam deploy` if you already have a `samconfig.toml`, or `sam deploy -g` if you don't. Additionally, there is a VS Code task that supports deploying various environments (staging, production, etc), which expects files called `samconfig.[stag|prod].toml`, and will use the correct one.
+See the [INSTALL](https://github.com/PRX/Porter/blob/master/INSTALL.md) guide for detailed information about deploying Porter to AWS.
 
-## Misc
+## State Machine Guidelines
 
 Various APIs and SDKs are inconsistent in how they refer to S3 buckets and objects. To be consistent throughout this project, they should be called `BucketName` and `ObjectKey` until they are passed to an API.
 
-For the sake of clarity, when building states always include the `InputPath`, `ResultPath`, and `OutputPath`, even if they are set to the default value.
+For the sake of clarity, when building states in ASL always include the `InputPath`, `ResultPath`, and `OutputPath`, even if they are set to the default value.
 
 Timestamps should always be in seconds.
 
-Create Lambda Layers for external dependencies, rather than including them in the deployment package.
+Create Lambda Layers for external dependencies. Do not include packages and libraries in the deployable Lambda code iteself.
 
-The job input format should generally avoid using input formats specific to particular tools or libraries. The underlying tools used to perform any given task could change over time.
+The job input format should generally avoid using parameters specific to particular tools or libraries. The underlying tools used to perform any given task could change over time. In some cases, such as with FFmpeg where recreating the entire API would be impractical, this rule can be adapted. In these cases, the input format should make it clear that the options or parameters are related specifically to one tool/library/etc.
```

Comment:
Yeah, I can imagine also passing direct flags in when doing video/image manipulation.